packages/playwright-test/src/index.ts

```diff
@@ -331,7 +332,10 @@ const playwrightFixtures: Fixtures<TestFixtures, WorkerFixtures> = ({
       (tracing as any)[startedCollectingArtifacts] = true;
       if (captureTrace) {
         let tracePath;
-        if (preserveTrace()) {
+        // Create a trace file if we know that:
+        // - it is's going to be used due to the config setting and the test status or
+        // - we are inside a test and the user manually closed the context.
+        if (preserveTrace() || runningTestBody()) {
```

Comment:
I guess we can also have a new context created and closed in `beforeEach`/`afterEach` hook, and we should preserve its trace as well? Perhaps just check whether we have started the fixture teardown phase or not.